### ISSUE TITLE
Results of Multi-task Learning. 

### ISSUE THREAD

**Issue by LeeShiyang** [THUDM/P-tuning-v2#22]
Thank you for providing the source code for the nice work. I have some questions regarding Multi-task Learning (MPT-2) in Table 4. In the paper, the authors mention that

>  **_For the multi-task setting, we combine the training set of the three datasets for pre-training. We use different linear classifiers for each dataset while sharing the continuous prompts._**. 

**_1. What do you mean by pre-training here? Do you have first pre-train on all datasets with labels and then continue to training on a specific dataset?
2. I have to find the pre-training code for multi-task learning in the repo but cannot find it. Is it possible that you put it public?_** 

Many thanks for the clarification!

**Comment by Xiao9905** [comment]
Hi @LeeShiyang ,
1. Your understanding is correct. We first pre-train our prompts on all datasets in a task, and then training on individual dataset.
2. This part of code may still need some time for us to prepare for publishing. It is also not difficult to implement. You can pass the dataset name into the data_loader, make sure all samples from one batch come from the same dataset, and use each dataset's unique linear head when predicting.

**Comment by LeeShiyang** [comment]
Thanks for the clarification!

**Comment by newtonysls** [comment]
你好，
1.请问这里的预训练是只预训练了prompts去初始化，还是连带的PLM的参数一起也变化了？
2.还有就是我在同样的3090上跑几次，RTE数据集的ACC最高只有76.5%和论文的SOTA差别很大，请问这些可能是什么原因呢？
感谢解答

**Comment by Xiao9905** [comment]
@newtonysls Hi,

1. 只训练了prompts，PLM的参数没有变化
2. 请问您配置的环境是按照我的README.md里的步骤命令和版本配置的吗？如果严格按照README.md的命令，结果应该是可复现的。

**Comment by newtonysls** [comment]
> 1. e first pre-train our prompts on al

python的版本是3.8.11
pytorch是1.10.0
torchvision是0.11.0
torchaudio是0.10.0
cuda是11.2
版本都是比readme上更新，这样也会影响的是嘛
虽然PLM没有变化，但是我看到@LeeShiyang的第一个问题，第一步带着标签把全部的数据进行一次prompt的预训练就相当于有效初始化，那这样其实是不是可以理解为前面也花了很多时间去预训练呢，因为在每层加入prompt还有forward和backward propagation，预训练速度也是非常慢的

**Comment by Xiao9905** [comment]
@newtonysls Hi,

1. 不同版本torch和cudatoolkit的确会造成影响。建议重新用conda按照README.md配置环境。
2. 你指的是multitask方法下的情况。论文中只在sequence tagging任务上使用了multitask（见表格中MPT-2列的结果），所有PT-2列的结果没有对prompt进行预训练

**Comment by newtonysls** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        
            
        
    
    十分感谢您的解答
        
    
     
    
        
                    ***@***.***
        
    
 

    On 5/9/2022 ***@***.***> wrote： 




@newtonysls Hi,



不同版本torch和cudatoolkit的确会造成影响。建议重新用conda按照README.md配置环境。

你指的是multitask方法下的情况。论文中只在sequence tagging任务上使用了multitask（见表格中MPT-2列的结果），所有PT-2列的结果没有对prompt进行预训练
